src/envoy/http/authn/authn_utils.cc

```diff
@@ -35,61 +35,47 @@ static const std::string kJwtIssuerKey = "iss";
 // The key name for the original claims in an exchanged token
 static const std::string kExchangedTokenOriginalPayload = "original_claims";
 
-// Extract JWT claim as a string list.
-// This function only extracts string and string list claims.
-// A string claim is extracted as a string list of 1 item.
-// A string claim with whitespace is extracted as a string list with each
-// sub-string delimited with the whitespace.
-void ExtractStringList(const std::string& key, const Envoy::Json::Object& obj,
-                       std::vector<std::string>* list) {
-  // First, try as string
-  try {
-    // Try as string, will throw execption if object type is not string.
-    const std::vector<std::string> keys =
-        absl::StrSplit(obj.getString(key), ' ', absl::SkipEmpty());
-    for (auto key : keys) {
-      list->push_back(key);
-    }
-  } catch (Json::Exception& e) {
-    // Not convertable to string
-  }
-  // Next, try as string array
-  try {
-    std::vector<std::string> vector = obj.getStringArray(key);
-    for (const std::string v : vector) {
-      list->push_back(v);
-    }
-  } catch (Json::Exception& e) {
-    // Not convertable to string array
-  }
-}
 };  // namespace
 
 bool AuthnUtils::ProcessJwtPayload(const std::string& payload_str,
                                    istio::authn::JwtPayload* payload) {
-  Envoy::Json::ObjectSharedPtr json_obj;
-  try {
-    json_obj = Json::Factory::loadFromString(payload_str);
-    ENVOY_LOG(debug, "{}: json object is {}", __FUNCTION__,
-              json_obj->asJsonString());
-  } catch (...) {
+  auto result = Wasm::Common::JsonParse(payload_str);
+  if (!result.has_value()) {
     return false;
   }
+  auto json_obj = result.value();
+  ENVOY_LOG(debug, "{}: json object is {}", __FUNCTION__, json_obj.dump());
 
   *payload->mutable_raw_claims() = payload_str;
 
   auto claims = payload->mutable_claims()->mutable_fields();
   // Extract claims as string lists
-  json_obj->iterate([json_obj, claims](const std::string& key,
-                                       const Json::Object&) -> bool {
-    // In current implementation, only string/string list objects are extracted
-    std::vector<std::string> list;
-    ExtractStringList(key, *json_obj, &list);
-    for (auto s : list) {
-      (*claims)[key].mutable_list_value()->add_values()->set_string_value(s);
-    }
-    return true;
-  });
+  Wasm::Common::JsonObjectIterate(
+      json_obj, [&json_obj, &claims](const std::string& key) -> bool {
+        // In current implementation, only string/string list objects are
+        // extracted
+        std::vector<std::string> list;
+        auto field_value =
+            Wasm::Common::JsonGetField<std::vector<std::string>>(json_obj, key);
```

Comment:
How about adding a GetField function which returns vector of string_view? Seems cheaper than vector of strings.

Reply:
done